src/shared/components/dateRangePicker/NewDatePicker.tsx

```diff
@@ -296,7 +326,7 @@ const DatePickerMenu: FC<Props> = ({onCollapse, timeRange, timeRangeLabel}) => {
                     : ComponentStatus.Error
                 }
                 type={InputType.Text}
-                value={inputStartDate ?? ''}
+                value={convertToDisplayFormat(inputStartDate, timeZone) ?? ''}
```

Comment:
The persisted time is always in UTC time zone, but users might have different setting with the time zone, either in "Local" or in "UTC". `convertToDisplayFormat()` convert the time to display according to the time zone